var/spack/repos/builtin/packages/py-fenics-basix/package.py

```diff
@@ -34,16 +35,18 @@ class PyFenicsBasix(PythonPackage):
     )
 
     depends_on("fenics-basix@main", type=("build", "run"), when="@main")
+    depends_on("fenics-basix@0.6.0", type=("build", "run"), when="@0.6.0")
     depends_on("fenics-basix@0.5.1", type=("build", "run"), when="@0.5.1")
     depends_on("fenics-basix@0.4.2", type=("build", "run"), when="@0.4.2")
     depends_on("fenics-basix@0.3.0", type=("build", "run"), when="@0.3.0")
     depends_on("fenics-basix@0.2.0", type=("build", "run"), when="@0.2.0")
     depends_on("fenics-basix@0.1.0", type=("build", "run"), when="@0.1.0")
 
+    depends_on("py-setuptools@42:", type=("build", "run"))
+    depends_on("py-numpy@1.2.1:", type=("build", "run"))
```

Comment:
1.21, not 1.2.1. Also, these requirements are only for the latest version, not all versions. You'll need to see when these requirements were introduced and add `when=` arguments as needed.

Reply:
NumPy version number fixed. 

`setuptools` and `numpy` have always been dependencies. Just the versioning in the Python packaging was a bit sloppy in the beta versions that were not on PyPI.

Reply:
So you're saying setuptools 42+, numpy 1.21+, python 3.8+, and pybind11 2.9.1+ have been required since 0.1.0? And that Python 3.7 has never worked?

Reply:
The would require trawling back over alpha and beta versions, and testing against all versions of dependencies. Hardly seems like a good use of time and effort, especially for deprecated versions.

Reply:
No need to test anything, I usually just use whatever was reported in pyproject.toml/setup.py/setup.cfg.

Reply:
Earlier versions did not include dependencies in `setup.py`, etc. Dependencies were cleaned up in recent versions when creating PyPI packages. Not sure how to move this forward.